crates/proof-of-sql/src/sql/postprocessing/group_by_postprocessing.rs

```diff
@@ -0,0 +1,322 @@
+            &min_columns,
+            &selection_column,
+        )?;
+        dbg!(&aggregation_results);
+        // Finally do another round of evaluation to get the final result
+        // Gather the results into a new OwnedTable
+        let group_by_outs: IndexMap<Identifier, OwnedColumn<S>> = aggregation_results
+            .group_by_columns
+            .iter()
+            .zip(self.group_by_identifiers.iter())
+            .map(|(column, id)| (*id, OwnedColumn::from(column)))
+            .collect::<IndexMap<_, _>>();
+        let sum_outs: IndexMap<Identifier, OwnedColumn<S>> = multizip((
+            aggregation_results.sum_columns.iter(),
+            sum_ids.iter(),
+            sum_column_types.iter(),
+        ))
+        .map(|(column, id, column_type)| {
+            Ok((*id, OwnedColumn::try_from_scalars(column, *column_type)?))
+        })
+        .collect::<PostprocessingResult<_>>()?;
+        let max_outs: IndexMap<Identifier, OwnedColumn<S>> = multizip((
+            aggregation_results.max_columns.iter(),
+            max_ids.iter(),
+            max_column_types.iter(),
+        ))
+        .map(|(column, id, column_type)| {
+            Ok((
+                *id,
+                OwnedColumn::try_from_option_scalars(column, *column_type)?,
+            ))
+        })
+        .collect::<PostprocessingResult<_>>()?;
+        let min_outs: IndexMap<Identifier, OwnedColumn<S>> = multizip((
+            aggregation_results.min_columns.iter(),
+            min_ids.iter(),
+            min_column_types.iter(),
+        ))
+        .map(|(column, id, column_type)| {
+            Ok((
+                *id,
+                OwnedColumn::try_from_option_scalars(column, *column_type)?,
+            ))
+        })
+        .collect::<PostprocessingResult<_>>()?;
+        //TODO: When we have NULLs we need to differentiate between count(1) and count(expression)
+        let count_column = OwnedColumn::BigInt(aggregation_results.count_column.to_vec());
+        let count_outs: IndexMap<Identifier, OwnedColumn<S>> = count_ids
+            .iter()
+            .map(|id| (**id, count_column.clone()))
+            .collect::<IndexMap<_, _>>();
+        let new_owned_table = OwnedTable::try_from_iter(
+            group_by_outs
+                .iter()
+                .chain(sum_outs.iter())
+                .chain(max_outs.iter())
+                .chain(min_outs.iter())
+                .chain(count_outs.iter())
+                .map(|(id, column)| (*id, column.clone())),
+        )?;
+        dbg!(&new_owned_table);
+        let result_index_map: IndexMap<Identifier, OwnedColumn<S>> = self
+            .remainder_exprs
+            .iter()
+            .map(|aliased_expr| {
+                let column = new_owned_table.evaluate(&aliased_expr.expr)?;
+                Ok((aliased_expr.alias, column))
+            })
+            .collect::<PostprocessingResult<_>>()?;
+        Ok(OwnedTable::try_new(result_index_map)?)
```

Comment:
@JayWhite2357 Since `OwnedTable::try_from_iter` has `T: IntoIterator<Item = (Identifier, OwnedColumn<S>)`
we can't use it unless we add a separate method to use `T: IntoIterator<Item = Result<(Identifier, OwnedColumn<S>, E>)`

Reply:
This should work:
```
let result = self
    .remainder_exprs
    .iter()
    .map(|aliased_expr| {
        let column = new_owned_table.evaluate(&aliased_expr.expr)?;
        PostprocessingResult::Ok((aliased_expr.alias, column))
    })
    .process_results(|iter| OwnedTable::try_from_iter(iter))??;
Ok(result)
```